src/test/client_test.ts

```diff
@@ -1747,3 +1747,58 @@ tap.only("VersionAllSomeFailed", function (t) {
     assertor(err);
   });
 });
+
+test("VersionAllWithErrorsSuccessful", function (t) {
+  const dummyServer1 = makeDummyVersionServer(t, "dummyServer1", "1.0.0");
+  const dummyServer2 = makeDummyVersionServer(t, "dummyServer2", "2.0.0");
+  const dummyServer3 = makeDummyVersionServer(t, "dummyServer3", "3.0.0");
+
+  const client = makeClient([dummyServer1, dummyServer2, dummyServer3]);
+  const assertor = function (
+    val?: Record<string, {version?: string | Buffer | null, error?: Error}> | null
+  ) {
+    t.deepEqual(
+      {
+        "dummyServer1:undefined": { version: "1.0.0" },
+        "dummyServer2:undefined": { version: "2.0.0" },
+        "dummyServer3:undefined": { version: "3.0.0" },
+      },
+      val
+    );
+  };
+
+  return client.versionAllWithErrors().then(function (res) {
+    assertor(res.values);
+  });
+});
+
+tap.only("VersionAllWithErrorsSomeFailed", function (t) {
+  const dummyServer1 = makeDummyVersionServer(t, "dummyServer1", "1.0.0");
+  const dummyServer2 = makeDummyVersionServer(t, "dummyServer2", "2.0.0");
+  dummyServer2.write = function () {
+    dummyServer2.error({
+      name: "ErrorName",
+      message: "This is an expected error.",
```

Comment:
Is there any real error example? 
Something like `EConnectError`?

Reply:
In the test we use a non-error object for ease of equality checking (though I suppose we could just construct an Error object of our choosing)

Possible errors are:

https://github.com/makenotion/memjs/blob/c84319eb3e955405ec82851f9c4d869382853d23/src/memjs/utils.ts#L159
https://github.com/makenotion/memjs/blob/c84319eb3e955405ec82851f9c4d869382853d23/src/memjs/server.ts#L168
https://github.com/makenotion/memjs/blob/c84319eb3e955405ec82851f9c4d869382853d23/src/memjs/server.ts#L231
https://github.com/makenotion/memjs/blob/c84319eb3e955405ec82851f9c4d869382853d23/src/memjs/server.ts#L251
https://github.com/makenotion/memjs/blob/c84319eb3e955405ec82851f9c4d869382853d23/src/memjs/server.ts#L326

any errors that socket can have (things like ECONNRESET, etc):
https://github.com/makenotion/memjs/blob/c84319eb3e955405ec82851f9c4d869382853d23/src/memjs/server.ts#L225-L227

and whatever this is

https://github.com/makenotion/memjs/blob/master/src/memjs/memjs.ts#L945